lib/fizzy/execute.cpp

```diff
@@ -758,27 +768,43 @@ execution_result execute(
                 goto end;
             }
 
-            const auto called_func_idx = (*instance.table)[elem_idx];
-            if (!called_func_idx.has_value())
+            const auto called_func = (*instance.table)[elem_idx];
+            if (!called_func.has_value())
             {
                 trap = true;
                 goto end;
             }
 
             // check actual type against expected type
-            const auto& actual_type = function_type(instance, *called_func_idx);
+            const auto& actual_type = called_func->type;
             const auto& expected_type = instance.module.typesec[expected_type_idx];
             if (expected_type != actual_type)
             {
                 trap = true;
                 goto end;
             }
 
-            if (!invoke_function(actual_type, *called_func_idx, instance, stack, depth))
+            const auto num_args = actual_type.inputs.size();
```

Comment:
Why did all this had to be move back here from within `invoke_function`?

Reply:
Duplication is resolved in the following commit